packages/backend/migrations/20231121184537_add_webhook_event_deposit_account_id.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function (knex) {
+  return knex.schema.alterTable('webhookEvents', function (table) {
+    table.uuid('depositAccountId').nullable()
+  })
+}
```

Comment:
I think the pros of individual columns are that it's more explicit and lets us use fkeys, but I dont like that there is nothing making them mutually exclusive (when they should be).  Whereas `resourceId` is simple and mutually exclusive (cant set both a incoming and outgoing payment resource id). However, it's ambiguous and not a fkey.

Not being a fkey _may_ be a red herring since we're not actually using the id to look up the related resource. Still, it kinda feels bad to store table id's without expressing the relationship. 

There is perhaps a more expressive way to model all of this. What about something like this? 

Add a new `OutgoingPaymentWebhookEvent` table like:

```sql
--  psuedoSQL
OutgoingPaymentWebhookEvent table {
  outgoingPaymentId: uuid fk referencing OutgoingPayment.id
  webhookEventId: uuid fk referencing WebhookEventId.id
}
```

Then we call something like `outgoingPaymentService.getLatestWebhookEventByType(outgoingPaymentId, "outgoing_payment.created")` which does something like this under the hood:

```
SELECT * 
FROM OutgoingPayment
JOIN OutgoingPaymentWebhookEvent on OutgoingPaymentWebhookEvent.outgoingPayment.id = OutgoingPayment.id
JOIN WebhookEvent on WebhookEvent.id = OutgoingPaymentWebhookEvent.webhookEventId
WHERE WebhookEvent.type = $1
ORDER BY WebhookEvent.created DESC
LIMIT 1;
```

Reply:
Pros and cons as I see it:

| strategy                | explicit | expresses mutual exclusivity | simple implementation | can understand resource type from webhookEvents table |
|-------------------------|----------|------------------------------|-----------------------|----------------------------------|
| seperate id colums      | 1        | 0                            | 1                     | 1                                |
| one "resourceId" column | 0        | 1                            | 1                     | 0                                |
| join table              | 1        | 1                            | 0                     | 0                                |

Reply:
In the separate id columns, we could also achieve mutual exclusivity by having some sort of check constraint that makes sure we have one of the separate id columns set.

I _do_ think the join table per type is probably the most robust solution, but I don't think it's necessarily worth it because we are not doing any look ups for the webhooks by peerId or assetId

Reply:
I agree that the join table is overkill. I think I prefer separate columns and a check that only one has been filled.

Reply:
I added the separate columns (`incomingPaymentId`, `outgoingPaymentId`, `walletAddressId`) with a constraint that only 1 is set. Didn't immediately see how `peerId` and `assetId` fit into this but I do now - adding those. 

One remaining detail - I think we need to set the associated resource id column for the existing `webhookEvents` to be thorough. They will all be `null`. Its possible this wont be a problem for finding them in the new resolvers, because integrators wont use them right away and the new `webhookEvents` will have them. But if there is some long lag between creating the webhook events and depositing/withdrawing (I dont think so?) then it would be a problem. 

Additionally, our constraint can't currently ensure there is at least 1 set, since existing records will have 0. So a new record could be added without any, which doesn't seem ideal.